tests/ops/test_channel_ops.py

```diff
@@ -137,6 +125,23 @@ def test_p_arbitrary(self, p, tol):
         expected_K1 = np.sqrt(p) * X
         assert np.allclose(op(p, wires=0).kraus_matrices[1], expected_K1, atol=tol, rtol=0)
 
+    @pytest.mark.parametrize("angle", np.linspace(0, 2 * np.pi, 7))
+    def test_grad_bitflip(self, angle):
+        """Test that analytical gradient is computed correctly for different states. Channel
+        grad recipes are independent of channel parameter"""
+
+        dev = qml.device("default.mixed", wires=1)
+        prob = 0.5
+
+        @qml.qnode(dev)
+        def circuit(p):
+            qml.RX(angle, wires=0)
+            qml.BitFlip(p, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        gradient = np.squeeze(circuit.jacobian(prob))
+        assert gradient == circuit(1)-circuit(0)
```

Comment:
Awesome!

It would be nice also to include an analytic check. That is, compute the analytic gradient as a function of `p`, and assert that the PL computed gradient matches this.

Reply:
The analytic gradient is independent of `p`! The only way we can test it is on different states. That's what's happening here: we prepare a different state, then check that the computed gradient coincides with the theoretical value

Reply:
Sorry, I had a typo in my comment. Meant to say 

> That is, compute the analytic gradient as a function of `angle`

Reply:
Josh made me do ALGEBRA and use TRIGONOMETRIC IDENTITIES and even COMPUTE TRACES BY HAND, but I added the tests :wink:

Reply:
I feel guilty 🙈